pkg/doc/jose/jwe.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+)
+
+// JSONWebEncryption represents a JWE as defined in https://tools.ietf.org/html/rfc7516.
+type JSONWebEncryption struct {
+	ProtectedHeaders   Headers
+	UnprotectedHeaders Headers
+	Recipients         []Recipient
+	AAD                string
+	IV                 string
+	Ciphertext         string
+	Tag                string
+}
+
+// Recipient is a recipient of a JWE including the shared encryption key
+type Recipient struct {
+	EncryptedKey string           `json:"encrypted_key,omitempty"`
+	Header       RecipientHeaders `json:"header,omitempty"`
+}
+
+// RecipientHeaders are the recipient headers
+type RecipientHeaders struct {
+	APU string `json:"apu,omitempty"`
+	IV  string `json:"iv,omitempty"`
+	Tag string `json:"tag,omitempty"`
+	KID string `json:"kid,omitempty"`
+	SPK string `json:"spk,omitempty"`
+}
+
+// rawJSONWebEncryption represents a RAW JWE that is used for serialization/deserialization.
+type rawJSONWebEncryption struct {
+	ProtectedHeaders   string          `json:"protected,omitempty"`
+	UnprotectedHeaders json.RawMessage `json:"unprotected,omitempty"`
+	Recipients         json.RawMessage `json:"recipients,omitempty"`
+	AAD                string          `json:"aad,omitempty"`
+	IV                 string          `json:"iv,omitempty"`
+	Ciphertext         string          `json:"ciphertext,omitempty"`
+	Tag                string          `json:"tag,omitempty"`
+}
+
+var errEmptyCiphertext = fmt.Errorf("ciphertext cannot be empty")
```

Comment:
We can use just `errors.New()` here.

Reply:
(since you are not formatting.)

Reply:
I debated between the two. errors.New adds an extra import, so that's why I went with fmt.errorf. Not sure if that's actually a real benefit, though. But I agree that errors.New looks clearer, especially since I'm not formatting.

I've changed it to errors.New :)